class.jetpack-gutenberg.php

```diff
@@ -937,4 +937,30 @@ public static function validate_block_embed_url( $url, $allowed = array(), $is_r
 		return false;
 	}
 
+	/**
+	 * Output an UpgradeNudge Component on the frontend of a site.
+	 *
+	 * @since 8.4.0
+	 *
+	 * @param string $plan The plan that users need to purchase to make the block work.
+	 *
+	 * @return string
+	 */
+	public static function upgrade_nudge( $plan ) {
+		/** This filter is documented in _inc/lib/admin-pages/class.jetpack-react-page.php */
```

Comment:
I find this comment a bit confusing.
The following line is not a filter, and the first filter appearing (`jetpack_block_editor_enable_upgrade_nudge`), although used in that file there, is in fact documented here (line 251)! 😄 

The comment seems to refer to `jetpack_show_promotions`, so I'd probably either move the comment down, or clarifiy it.

Reply:
Yeah, this isn't going to work I am afraid, the parser won't pick it up. We'd need to move the comment right above its filter. We'll need a similar one for `jetpack_block_editor_enable_upgrade_nudge`:

`/** This filter is documented in class.jetpack-gutenberg.php */`

(even if we're in the same file. That will help the parser understand that no page should be created for that hook).

Something like this:

```diff
diff --git a/class.jetpack-gutenberg.php b/class.jetpack-gutenberg.php
index 577fa4e8b..d7759ec8c 100644
--- a/class.jetpack-gutenberg.php
+++ b/class.jetpack-gutenberg.php
@@ -947,11 +947,14 @@ class Jetpack_Gutenberg {
 	 * @return string
 	 */
 	public static function upgrade_nudge( $plan ) {
-		/** This filter is documented in _inc/lib/admin-pages/class.jetpack-react-page.php */
-		if ( ! current_user_can( 'manage_options' )
+		if (
+			! current_user_can( 'manage_options' )
+			/** This filter is documented in class.jetpack-gutenberg.php */
 			|| ! apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false )
+			/** This filter is documented in _inc/lib/admin-pages/class.jetpack-react-page.php */
 			|| ! apply_filters( 'jetpack_show_promotions', true )
-			|| is_feed() ) {
+			|| is_feed()
+		) {
 			return;
 		}
 

```